runtime/major_gc.c

```diff
@@ -241,7 +241,16 @@ Caml_inline void mark_stack_push(struct mark_stack* stk, value block,
   for (i = offset; i < end; i++) {
     v = Field(block, i);
 
-    if (Is_block(v) && !Is_black_val(v))
+    if (Is_block(v) &&
+#ifdef NO_NAKED_POINTERS
+        /* With NO_NAKED_POINTERS, every block has a valid header. */
+        !Is_young(v) &&
```

Comment:
The original code did not ignore pointers to the minor heap.  In other words, if a small major heap block contains only pointers to the minor heap, it is now skipped (not pushed on the mark stack), while it was pushed on the mark stack before.  That is probably correct, but I'd like to be sure.

Also, the comment doesn't match the test that follows.  If "every block has a valid header", why check `!Is_young` ?

Reply:
> The original code did not ignore pointers to the minor heap. In other words, if a small major heap block contains only pointers to the minor heap, it is now skipped (not pushed on the mark stack), while it was pushed on the mark stack before. That is probably correct, but I'd like to be sure.

This is correct.